fix: Py dev guide Update (#252)
GreptimeTeam/docs: docs/user-guide/python-coprocessor/overview.md

* [Builtin Module](./builtins.md)
* [Third-parties Libs](./third-parties.md)
* [FAQ](./faq.md)

# Note:

> The python coprocessor is still in the experimental stage, and the API is subject to change. 

Reviewer: ```suggestion
 The Python coprocessor is currently in its experimental phase, and the API may undergo some changes.
```

> It also comes with some shenanigans, you need to set up the path of correct python shared library, this could a bit tricky, usually you just need to install `python-dev` package(Like on most Debian-based system), but on macOS you might need to set up a correct soft link to `Library/Frameworks/Python.framework` if you are using homebrew to install python.

Reviewer: ```suggestion
using it also involves some complications. You must set up the correct Python shared library, which can be a bit challenging. In general, you just need to install the `python-dev` package(on most Debian-based system). However, if you are using Homebrew to install Python on macOS, you must create a proper soft link to `Library/Frameworks/Python.framework`.
```

> So if you can't run your binary on macOS because of linking error, you can try to fix it as below:

Reviewer: ```suggestion
So if you can't run your binary on macOS due to a linking error, you can try to fix it as below:
```

> So if you can't run your binary on MacOs because of link error, you can try to fix it like this(This is a temporary solution, we will fix it in the future):

Reviewer 2: ```suggestion
So if you can't run your binary on macOS because of link error, you can try to fix it like this (This is a temporary solution, we will fix it in the future):
```

> So if you can't run your binary on MacOs because of link error, you can try to fix it like this(This is a temporary solution, we will fix it in the future):

Reviewer 3: ```suggestion
So if you can't run your binary on macOS because of linking error, you can try to fix it as below:
```

> 1. Open your python and run the following code to get your python shared library path:

Reviewer: ```suggestion
1. Open your Python and run the following code to get your Python shared library path:
```

```python
from sysconfig import get_config_var, get_platform
get_config_var("LIBDIR")
```
This should give you something like `'/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/lib'`.

> 2. create a config file named `pyo3.config` in the same directory of your binary, and add the following content:

Reviewer: ```suggestion
2. Create a config file named `pyo3.config` in the same directory of your binary, and add the following content:
```

```config
implementation=CPython
version=3.9
shared=true
abi3=false
lib_name=python3.9
lib_dir=<Your LIB_DIR get from step 1>
executable=<You python executable path, could come from `which python`>
pointer_width=64
build_flags=
suppress_build_script_link_lines=false
```

> You might also want to change the `version` and `lib_name` to match your python version.

Reviewer: ```suggestion
You might also need to change the `version` and `lib_name` to correspond with your specific version of Python.
```
